### ISSUE TITLE
hi, can someone please advise me how to get 42coin working

### ISSUE THREAD

**Issue by david739** [oliverw/miningcore#1582]
hi, can someone please advise me how to get 42coin working
my config
rpcuser=xxx
rpcpassword=xxx
rpcport=2121
daemon=1
server=1
rpcallowip=127.0.0.1

and my coins.json has
"42": {
"name": "42",
"symbol": "42",
"family": "Bitcoin",
"coinbaseHasher": {
"hash": "sha256d"
},
"headerHasher": {
"hash": "scrypt",
"args": [ 1024, 1 ]
},
"blockHasher": {
"hash": "reverse",
"args": [ { "hash": "sha256d" } ]
},
"explorerBlockLink": "https://chainz.cryptoid.info/42/",
"explorerTxLink": "https://chainz.cryptoid.info/42/tx.dws?",
"explorerAccountLink": "https://chainz.cryptoid.info/42/address.dws?"
},

config.json
"id": "42",
"enabled": true,
"coin": "42",
"address": "xxx",
"rewardRecipients": [
{
"address": "xxx",
"percentage": 1.0
}
],
"blockRefreshInterval": 500,
"jobRebroadcastTimeout": 10,
"clientConnectionTimeout": 600,
"banning": {
"enabled": true,
"time": 600,
"invalidPercent": 50,
"checkThreshold": 50
},
"ports": {
"3021": {
"listenAddress": "0.0.0.0",
"difficulties": 1024,
"varDiff": {
"minDiff": 512,
"targetTime": 15,
"retargetTime": 90,
"variancePercent": 30
}
}
},
"daemons": [
{
"guest": "127.0.0.1",
"port": 2121,
"user": "xxx",
"password": "xxx"

still writes [42] Waiting for daemons to come online ...

**Comment by david739** [comment]
hello, I have a problem with ETC, it keeps writing to me on the miner (consecutive shared zamita pool) the coin is displayed and the work goes on, but the hash of the miner is not displayed on the website

**Comment by pool2mine** [comment]
Not 100% sure, but as far as i remember, Miningcore has no support for ETC yet. Remember ETC hardforked and changed algo a couple of years ago to ETCHASH

**Comment by david739** [comment]
hi i have etc. the miner receives it, but the hash is not displayed on the server and the work is not confirmed

**Comment by pool2mine** [comment]
I never got it working with miningcore... I had to fork the sammy's version that official ETC profile has, then added nicehash support and some other custom stuff i needed. it is easier to start with their repo as they already added ETCHASH support

**Comment by jon4hz** [comment]
#1514 adds support for ETC (not merged yet though)

**Comment by david739** [comment]
hello, I copied the data and checked the source, it should fit according to the instructions, but I still have something wrong, can someone please help me

Generating DAG for epoch 562


[2023-01-29 19:34:04.3201] [I] [etc1] Broadcasting job 00000073 
[2023-01-29 19:34:05.3469] [I] [StatsRecorder] [etc1] Updating Statistics for pool 
[2023-01-29 19:34:05.3890] [I] [StatsRecorder] [etc1] Reset performance stats for pool 
[2023-01-29 19:34:08.3298] [I] [etc1] [0HMO24O99RQ53] Share rejected: low difficulty share (2.3283064365386963E-10) [] 
[2023-01-29 19:34:10.3315] [I] [etc1] New work at height 16838286 and header 0xc5b7438db8c4676eae6be51f5219eaa3973bb03bb90648b7b86147af3ff9323b via [POLL] 
[2023-01-29 19:34:10.3315] [I] [etc1] Broadcasting job 00000074 
[2023-01-29 19:34:11.5124] [I] [etc1] [0HMO24O99RQ53] Share rejected: low difficulty share (4.656612873077393E-10) [] 
[2023-01-29 19:34:11.5163] [I] [etc1] [0HMO24O99RQ53] Connection closed 
[2023-01-29 19:34:16.5339] [I] [etc1] [0HMO24O99RQ54] Accepting connection from ::ffff:192.168.0.1:50232 ... 
[2023-01-29 19:34:16.5339] [I] [etc1] [0HMO24O99RQ54] Connection from ::ffff:192.168.0.1:50232 accepted on port 4001 





Pool coin	Algorithm	Miners	Pool Hashrate	Fee	Network Hashrate	Network Difficulty	
Ethereum Classic (ETC)	Ethhash	0	0 H/s	1 %	125.86995 TH/s	1.64834 PGO MINE ETHEREUM CLASSIC

**Comment by jon4hz** [comment]
That's definitely the wrong epoch. What chaintypeoverride did you use in the config.json?

**Comment by david739** [comment]
![1](https://user-images.githubusercontent.com/46352851/215349039-ab1f6476-88d2-4d6c-87e3-838c5628b46d.jpg)

**Comment by david739** [comment]
hello, I didn't change config.json, there is no information in the instructions, what should I change there, please

**Comment by jon4hz** [comment]
try to add `"chainTypeOverride": "Classic"` in the pool config.

**Comment by david739** [comment]
hi you mean like this
config.chceme.json

902
    "chainTypeOverride": "Classic" {
      "type": [
        "integer",
        "null"
      ]
    },
    "equihashMaxThreads": {
      "type": [
        "integer",
        "null"
      ]
    },
    "instanceId": {
      "type": [
        "integer",
        "null"
      ]
    },
    "logging": {
      "$ref": "#/definitions/ClusterLoggingConfig"

**Comment by jon4hz** [comment]
no, your pool config...

**Comment by david739** [comment]
sorry i got it here is my settings

`{
"logging": {
"level": "info",
"enableConsoleLog": true,
"enableConsoleColors": true,
"logFile": "bitcoin-cash.log",
"apiLogFile": "bitcoin-cash_api.log",
"logBaseDirectory": "c:/miningcore/build/logs",
"perPoolLogFile": false
},
"banning": {
"manager": "Integrated",
"banOnJunkReceive": false,
"banOnInvalidShares": false
},
"notifications": {
"enabled": false,
"email": {
"host": "smtp.example.com",
"port": 587,
"user": "user",
"password": "password",
"fromAddress": "info@yourpool.org",
"fromName": "pool support"
},
"admin": {
"enabled": false,
"emailAddress": "user@example.com",
"notifyBlockFound": false
}
},
"persistence": {
"postgres": {
"host": "127.0.0.1",
"port": 5432,
"user": "miningcore",
"password": "xxx",
"database": "xxx"
}
},
"paymentProcessing": {
"enabled": true,
"interval": 900,
"shareRecoveryFile": "recovered-shares.txt"
},
"api": {
        "enabled": true,
        "listenAddress": "*",
        "port": 4000,
        "metricsIpWhitelist": [],
        "rateLimiting": {
            "disabled": true,
            "rules": [
                {
                    "Endpoint": "*",
                    "Period": "1s",
                    "Limit": 5
                }
            ],
            "ipWhitelist": [
                ""
            ]
        }
    },
    "clusterName": "eu1",
    "pools": [
      {
        "id": "etc1",
        "enabled": true,
        "coin": "ethereumclassic",
        "address": "xxx",
        "rewardRecipients": [
          {
            "type": "op",
            "address": "xxx",
            "percentage": 1.0
          }
        ],
        "blockRefreshInterval": 120,
        "clientConnectionTimeout": 600,
        "banning": {
          "enabled": true,
          "time": 600,
          "invalidPercent": 50,
          "checkThreshold": 50
        },
        "ports": {
          "4001": {
            "name": "GPU-SMALL",
            "listenAddress": "*",
            "difficulty": 0.1,
            "varDiff": {
              "minDiff": 0.1,
              "maxDiff": null,
              "targetTime": 15,
              "retargetTime": 90,
              "variancePercent": 30
            }
          }
        },
        "chainTypeOverride": "Classic",
        "dagDir": "<Your directory for the dag file>",
        "daemons": [
          {
            "host": "127.0.0.1",
            "port": 8545,
            "user": "",
            "password": ""
          }
        ],
        "paymentProcessing": {
            "enabled": true,
            "minimumPayment": 0.1,
            "payoutScheme": "PPLNS",
            "payoutSchemeConfig": {
                "factor": 0.5
            },
            "gas": 21000,
            "maxFeePerGas": 50000000000,
            "BlockSearchOffset": 100,
            "keepUncles": false,
            "keepTransactionFees": false
        }
      }
    ]
}`

**Comment by jon4hz** [comment]
Are you running on windows? (looking at the `logBaseDirectory`)
Not sure if windows is fully supported.

**Comment by david739** [comment]
i have ubuntu it's just a bug so what now

**Comment by jon4hz** [comment]
Maybe @blackmennewstyle can help you out. He wrote that code :)

**Comment by david739** [comment]
ok thank you
and it won't be bad (Generating DAG for epoch 562)

**Comment by blackmennewstyle** [comment]
Hi,

Sorry for the late reply. Generally when i faced this error, it was because of the DAG path. Is your path not filled?

`"dagDir": "<Your directory for the dag file>",`

**Comment by blackmennewstyle** [comment]
> Hi,
> 
> Sorry for the late reply. Generally when i faced the error `Share rejected: low difficulty share`, it was always because of the `DAG`. Is your path not filled?
> 
> `"dagDir": "<Your directory for the dag file>",`
> 
> Do not hesitate to read that discussion: #1586

**Comment by david739** [comment]
hi, I'll take a look at it, but I don't have a bad trigger for getg yet

./geth --classic --maxpeers "350" --cache "8192" --allow-insecure-unlock --miner.etherbase "xxx" --unlock "xxx" --password ~/pwd --mine --http --syncmode "fast" --http.api "admin,debug,web3,eth,txpool,personal,ethash,miner,net,trace"

**Comment by blackmennewstyle** [comment]
The POW algorithms `etchash` needs the DAG files located in that directory in order to work properly and validate the shares it receives from miners. If you does not specify the path for that directory in your `config.json`, your shares will always be rejected.

Also, what do you mean by `bad trigger` with `geth`?

**Comment by david739** [comment]
that will be it can you please provide an example in the source code

**Comment by david739** [comment]
I already put it in there, but the file classic.toml is empty, does anyone know what the source should be in it
      
"chainTypeOverride": "Classic",
"dagDir": "/home/xxx/.ethereum/classic/geth/classic.toml",
"daemons": [

**Comment by blackmennewstyle** [comment]
Out of curiosity, are you trying to run your ETC pool with the source code currently available on `https://github.com/oliverw/miningcore`?

**Comment by blackmennewstyle** [comment]
I feel like i misunderstood you since beginning, so my bad.

So like @jon4hz warned about, the source code currently available on `https://github.com/oliverw/miningcore` does not support ETC with `etchash`, it has not been merged yet. So if you want to have support for ETC with `etchash`, you need to use my fork `https://github.com/blackmennewstyle/miningcore`.

Also, you need clearly to read the `geth` documentation in order to be more familiar. The `.toml` file is a configuration file for `geth`, you need to generate it. It allows you to run the `node` without the need of specifying one hundreds options.
Once you generated that `.tonl` file, it will contain a line like this one `DatasetDir = "/home/ceedii/.etchash"`. It's the path to the directory which contains all the DAG files, it's the one you can use for `"dagDir":`.
However, if you are in a situation where `miningcore` and `geth` are not running on the same server. You can arbitrarily create a folder on the server which runs `miningcore` and specify the path to that folder inside `"dagDir":`. `miningcore` will automically generated the necessary DAG files. Just make sure that the user which runs `miningcore` has `read` and `write` access on that folder.

**Comment by david739** [comment]
ok i will do everything and let you know

**Comment by david739** [comment]
and could you please put the source of your .toml file here, I would use it as a template

**Comment by david739** [comment]
I would be grateful for the procedure how to generate .toml

**Comment by jon4hz** [comment]
I don't think that is neccessary at all. `dagDir`  is an optional parameter (or at least was until now). 

What stands out is that your pool tried to generate the dag for epoch 562 which is wrong. Etc is on epoch 280. 
But that should have anything to do with the directoy where the dag is stored.

**Comment by david739** [comment]
ok, and how should I generate 280, it still generates 562, it wouldn't be possible to fix something

./geth --classic --maxpeers "350" --cache "8192" --allow-insecure-unlock --miner.etherbase "xxx" --unlock "xxx" --password ~/pwd --mine --http --syncmode "fast" --http.api "admin,debug,web3,eth,txpool,personal,ethash,miner,net,trace"

**Comment by blackmennewstyle** [comment]
Please, can you answer which source code you are currently using? `https://github.com/oliverw/miningcore` or `https://github.com/blackmennewstyle/miningcore` for running your pool?

**Comment by david739** [comment]
yeah sorry I'm trying yours now https://github.com/blackmennewstyle/miningcore

**Comment by david739** [comment]
tak na tom vašem mam 280 teď to nabíhá tak uvidíme

**Comment by david739** [comment]
so your system works without a problem thank you thank you and 1 more thank you

**Comment by blackmennewstyle** [comment]
You're welcome. So you were just not using the correct source code. Congratulations :)
